driver-async/src/main/com/mongodb/async/client/OperationIterable.java

```diff
@@ -117,6 +117,12 @@ public void onResult(final List<T> results, final Throwable t) {
         return new MappingIterable<T, U>(this, mapper);
     }
 
+    @Override
+    public OperationIterable<T> batchSize(final int batchSize) {
```

Comment:
I think this should throw.  Reason being that it's a private class and as such the driver should never be calling this method, and if it does, it's a bug (like the one you fixed in the last commit).